### ISSUE TITLE
Tab complete changed in inconvenient way for PS7 Preview3

### ISSUE THREAD

**Issue by rkeithhill** [PowerShell/PowerShell#10482]
Tab-complete is finding files from somewhere (not sure where).

![PS7AutoCompleteBroken](https://user-images.githubusercontent.com/5177512/64272073-d197a400-cefb-11e9-947a-e163b655d717.gif)

# Steps to reproduce

```powershell
1. dotnet new console -o HelloWorld
2. dotnet run
3. bi<tab>
4. <tab>
```

# Expected behavior

```none
Complete to `.\bin\Debug`
```

# Actual behavior

```none
completes to `B:` or other odd files:
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0-preview.3
PSEdition                      Core
GitCommitId                    7.0.0-preview.3
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by vexx32** [comment]
It looks kinda like it's tab-completing from `$env:PATH`? I don't remember it doing that before... I don't think it's necessarily a negative, that's somewhat helpful. The `B:` entry is an odd one, wonder where that cropped up from?

In any case, even the ones that might be useful here should probably end up at the bottom of the tab complete list, below any PowerShell commands (I notice Pester's BeforeAll is somewhere in the middle of the list, below several of the extraneous entries, for example).

**Comment by rkeithhill** [comment]
Or in this case, below any **local** files and directories.

**Comment by iSazonov** [comment]
Does it work well in 7.0.0-preview.2? Is it only Unix regression?

**Comment by vexx32** [comment]
Both @rkeithhill and I are seeing this on Windows. Not sure about other platforms. :)

**Comment by iSazonov** [comment]
Thanks! I can repo on Windows. On 6.2.2 too. So it is not regression in 7.0. (I can repo without PSReadline too).

**Comment by iSazonov** [comment]
Although possibly it is not bug - for `.\b<tab>` I get extected results. I guess `b<Tab>` falls in last resort code path and it is expected behavior.

**Comment by TravisEz13** [comment]
I think honestly I regressed the 6.2 behavior in making a security fix and then restored the old behavior when an issue was filed for a regression and said here https://github.com/PowerShell/PowerShell/issues/10482#issuecomment-528385612

I can look up the changes.

**Comment by TravisEz13** [comment]
These are the changes:
https://github.com/PowerShell/PowerShell/pull/9202
https://github.com/PowerShell/PowerShell/pull/9623

**Comment by TravisEz13** [comment]
#9623 fixed https://github.com/PowerShell/PowerShell/issues/9396 which dealt with drive like cases.

**Comment by ThomasNieto** [comment]
Should this be fixed for 7.1 GA?
